src/main/java/seedu/address/logic/parser/AddModuleClassCommandParser.java

```diff
@@ -0,0 +1,48 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.AddModuleClassCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.components.Name;
+import seedu.address.model.moduleclass.ModuleClass;
+
+/**
+ * Parses input arguments and creates a new AddModuleClassCommand object.
+ */
+public class AddModuleClassCommandParser implements Parser<AddModuleClassCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddStudentCommand
+     * and returns an AddStudentCommand object for execution.
```

Comment:
I think this should change to `AddModuleClassCommand`?
```suggestion
     * Parses the given {@code String} of arguments in the context of the AddModuleClassCommand
     * and returns an AddModuleClassCommand object for execution.
```

Reply:
Oops, that slipped through my manual search and replace.